src/test/java/com/nashtech/controller/ReactiveDataControllerTest.java

```diff
@@ -0,0 +1,122 @@
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+
+@ExtendWith(MockitoExtension.class)
+public class ReactiveDataControllerTest {
+
+    @Mock
+    private ReactiveDataService reactiveDataService;
+
+    @InjectMocks
+    private ReactiveDataController reactiveDataController;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void testGetAllBrands_ReactiveDataServiceReturnsError() {
+        // Setup
+        when(reactiveDataService.getAllBrands()).thenReturn(Flux.error(new Exception("message")));
+
+        // Run the test
+        final Flux<CarBrand> result = reactiveDataController.getAllBrands();
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectErrorMatches(throwable -> throwable instanceof Exception && throwable.getMessage().equals("message"))
+                .verify();
+    }
+
+    @Test
+    void testGetAllBrands_ReactiveDataServiceReturnsNoItem() {
+        // Setup
+        when(reactiveDataService.getAllBrands()).thenReturn(Flux.empty());
+
+        // Run the test
+        final Flux<CarBrand> result = reactiveDataController.getAllBrands();
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+    @Test
+    void testGetCarsByBrand_ReactiveDataServiceReturnsError() {
+        // Setup
+        when(reactiveDataService.getCarsByBrand("brand")).thenReturn(Flux.error(new Exception("message")));
+
+        // Run the test
+        final Flux<Car> result = reactiveDataController.getCarsByBrand("brand");
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectErrorMatches(throwable -> throwable instanceof Exception && throwable.getMessage().equals("message"))
+                .verify();
+    }
+
+    @Test
+    void testGetCarsByBrand_ReactiveDataServiceReturnsNoItem() {
+        // Setup
+        when(reactiveDataService.getCarsByBrand("brand")).thenReturn(Flux.empty());
+
+        // Run the test
+        final Flux<Car> result = reactiveDataController.getCarsByBrand("brand");
+
+        // Verify the results
+        StepVerifier.create(result)
+                .expectNextCount(0)
+                .verifyComplete();
+    }
+
+
+    @Test
+    void testPushDataToCloud() {
+        ResponseEntity<Object> response = reactiveDataController.pushDataToCloud();
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+        verify(reactiveDataService).fetchAndSendData();
+    }
+
+    @Test
+    void testGetCarsByBrand() {
+        String brand = "Toyota";
+        when(reactiveDataService.getCarsByBrand(brand)).thenReturn(Flux.just(new Car(), new Car()));
+
+        Flux<Car> carsFlux = reactiveDataController.getCarsByBrand(brand);
+
+        StepVerifier.create(carsFlux)
+                .expectNextCount(2)
```

Comment:
You are simply checking the count here, but you have to check whether the returned car is of same brand or not e.g. Toyota.